packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -391,7 +404,25 @@ export function TransferPanel() {
 
             await latestToken.current.approve({
               erc20L1Address: selectedToken.address,
-              l1Signer: latestNetworksAndSigners.current.l1.signer
+              l1Signer: latestNetworksAndSigners.current.l1.signer,
+              txLifecycle: {
+                onTxSubmit: ({ tx, symbol }) => {
+                  transactions.addTransaction({
+                    type: 'approve',
+                    status: 'pending',
+                    value: null,
+                    txID: tx.hash,
+                    assetName: symbol,
+                    assetType: AssetType.ERC20,
+                    sender: walletAddress,
+                    l1NetworkID
+                  })
+                },
+                onTxSuccess: ({ tx, txReceipt }) => {
+                  transactions.updateTransaction(txReceipt, tx)
+                },
+                onTxFailure: txHash => txLifecycleOnTxFailure(txHash)
```

Comment:
@spsjvc After a closer look into the code, I think other than for `triggerOutbox` methods, we should name the 2nd lifecycle method `onTxConfirm`
I don't think we should have `onTxSuccess` and `onTxFailure` because `transactions.updateTransaction` checks the tx receipt status to call `transactions.setTransactionSuccess` or `transactions.setTransactionFailure`.

https://github.com/OffchainLabs/arb-token-bridge/blob/0c060428a4d5c6fad7dbe2129d3c3d2a577f957d/packages/token-bridge-sdk/src/hooks/useTransactions.ts#L460-L468

it seems to me that the `triggerOutbox` methods for both ETH and tokens needed to call methods different from what's done in `updateTransaction` so `setTransactionSuccess` and `setTransactionFailure` are called instead of `updateTransaction`